src/main/java/Bob/Parser.java

```diff
@@ -0,0 +1,137 @@
+package bob;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.List;
+import java.util.Locale;
+
+import bob.task.Deadline;
+import bob.task.Event;
+import bob.task.Task;
+import bob.task.Todo;
+
+/**
+ * Represents a parser that parses user input and executes the corresponding
+ * commands.
+ */
+class Parser {
+
+    /**
+     * Parses a string into a LocalDateTime object.
+     *
+     * @param input
+     * @return LocalDateTime
+     * @throws BobException
+     */
+    public static LocalDateTime parseDateTime(String input) throws BobException {
+        try {
+            // Correct format: "dd/MM/yyyy HHmm" (e.g., "02/12/2019 1800")
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d/M/yyyy HHmm", Locale.ENGLISH);
+            return LocalDateTime.parse(input, formatter);
+        } catch (DateTimeParseException e) {
+            throw new BobException("The date and time format is incorrect: " + e.getMessage());
+        }
+    }
```

Comment:
This method seems out of place consider the purpose of `Parser`. Perhaps if it is not used anywhere else you can make it private? It is more of a helper function to `Parser` as compared to something that `Parser` does that other classes can use. The main thing `Parser` seems to do is to interpret and execute commands, not to parse `DateTime`s.

Alternatively, move it to another class